Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -89,47 +89,30 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
-    /// Holds the date when the status was last changed so we can send it out as additional information
-    /// in our status-change notifications.
-    ///
-    private var lastStatusChangeDate = Date()
-
-    private var connectionStatus: ConnectionStatus = .disconnected {
+    public var connectionStatus: ConnectionStatus = .disconnected {
         didSet {
-            if oldValue != connectionStatus {
-                lastStatusChangeDate = Date()
-                broadcastConnectionStatus()
+            guard connectionStatus != oldValue else {
+                return
             }
+
+            connectionStatusPublisher.send(connectionStatus)
         }
     }
 
-    private func broadcastConnectionStatus() {
-        let lastStatusChange = ConnectionStatusChange(status: connectionStatus, on: lastStatusChangeDate)
-        let payload = ConnectionStatusChangeEncoder().encode(lastStatusChange)
+    public let connectionStatusPublisher = CurrentValueSubject<ConnectionStatus, Never>(.disconnected)
```

Comment:
`CurrentValueSubject` publishes the last known value to any new subscriber, and any updates after the subscription is started.